### PR TITLE
feat: connector capability markers (location shape/identity/record-version)

### PR DESCRIPTION
**Pull request by tabossert**
## Connector capability markers (location shape / identity / record-version)

Adds capability **markers** to the connector registry so the control plane (platform-api / platform-etl) can derive per-connector parameterization behavior instead of hardcoding it.

**What's here**
- `LocationShape` enum + kw-only markers on `SourceRegistryEntry` / `DestinationRegistryEntry`:
  `location_shape`, `location_identity`, `supports_recursion`, `emits_record_version`
- Markers default to `None` (unannotated) so consumers fall back to their hand-seeded table for connectors not yet annotated — no behavior change until each entry is explicitly marked

Part of the parameterized-connectors generalization (consumed by platform-api #1210/its follow-up and platform-etl).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<a href="https://cubic.dev/pr/Unstructured-IO/unstructured-ingest/pull/752?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: unstructured_ingest/processes/connectors/zendesk/zendesk.py [hunk 1]
```diff
@@ -238,4 +245,7 @@ async def run_async(self, file_data: FileData, **kwargs: Any) -> DownloadRespons
     indexer=ZendeskIndexer,
     downloader=ZendeskDownloader,
     downloader_config=ZendeskDownloaderConfig,
+    location_shape=LocationShape.API_FOLDER,
+    location_identity=("connector_config.subdomain", "indexer_config.item_type"),
```
[the remark this answers is not in the document]
Comment (tabossert) [reply unstructured_ingest/processes/connectors/zendesk/zendesk.py:249]: Declining — this is the intended convention, not a bug. In `location_identity`, the `connector_config.*` root maps to the connector's `ConnectionConfig`: `test_registry_markers.py::test_annotated_identity_leaves_match_eligible_markers` resolves `connector_config` → the entry's `connection_config`. `subdomain` is a real field on `ZendeskConnectionConfig`, so `connector_config.subdomain` is correct. Renaming to `connection_config.subdomain` would introduce a root that does not exist and fail that invariant test.

File: unstructured_ingest/processes/connectors/elasticsearch/elasticsearch.py [hunk 2]
```diff
@@ -467,6 +469,10 @@ def _sanitize_bulk_index_error(self, error: dict[str, dict]) -> dict:
     indexer_config=ElasticsearchIndexerConfig,
     downloader=ElasticsearchDownloader,
     downloader_config=ElasticsearchDownloaderConfig,
+    location_shape=LocationShape.SEARCH_INDEX,
+    location_identity=("connector_config.hosts", "indexer_config.index_name"),
```
[the remark this answers is not in the document]
Comment (tabossert) [reply unstructured_ingest/processes/connectors/elasticsearch/elasticsearch.py:502]: Declining on the source — the guarded collision is unreachable here. `ElasticsearchIndexer.run` dereferences `self.connection_config.hosts[0]` unconditionally, so a `cloud_id`-only source cannot run; `hosts` is de-facto required and always present in the source identity, so adding an always-`None` `cloud_id` leaf adds no discrimination. The **destination** path does use `hosts or cloud_id`, so `cloud_id` was added to the destination identity instead (commit bbfd9750).

File: test/unit/connectors/test_registry_markers.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+def _eligible(config) -> set[str]:
+    props = config.model_json_schema().get("properties", {})
+    return {name for name, prop in props.items() if prop.get("x-runtime-eligible") is True}
+
+
+def _fields(config) -> set[str]:
+    return set(config.model_json_schema().get("properties", {}).keys())
+
+
```
Comment (potter-potter) [inline test/unit/connectors/test_registry_markers.py:208]: This test only checks entries that are annotated. It'd be worth adding a test for the opposite case too: that entries with location_shape=None are treated as not parameterizable. That locks in the "unannotated = ignored" behavior so it can't silently break later.

**Review by potter-potter — APPROVED** [review]
Looks good. If you can just add the extra tests from PR comment

File: test/unit/connectors/test_registry_markers.py (hunk 3 above)
Comment (tabossert) [reply test/unit/connectors/test_registry_markers.py:208]: Added `test_unannotated_entries_are_ignored` (parametrized over source + destination) in `f443740a`.

Worth noting a subtlety I hit: the obvious version — "unannotated entries expose no `x-runtime-eligible` fields" — actually fails, because several unannotated entries (`databricks_volumes_*`, `sqlite`, `qdrant-*`, `weaviate-*`) inherit eligible fields from a config class they share with an annotated sibling. So the test instead locks in the real invariant: those entries keep the default fsspec-shaped `location_identity` and default markers, yet stay `location_shape=None` and are ignored — proving parameterization is gated **solely** on `location_shape`, not on the presence of eligible fields. That's exactly the "unannotated = ignored" behavior you wanted guarded.
